### PR TITLE
DAOS-10249 server: Fix uninitialized node_id

### PR DESCRIPTION
**Pull request by liw**
The node_id field of raft_server_private_t is initialized to 0 by
raft_new. Since 0 may be a valid node ID, raft_new should initialize
node_id to -1 instead, like leader_id and what raft_clear does to
node_id. This patch fixes the node_id initialization and adds a new API
function, raft_set_nodeid, for users to set the node ID. Users shall
call raft_set_nodeid right after raft_new/raft_clear. (Adding a node ID
parameter to raft_new/raft_clear would be a better solution, though it
would require a lot of test changes.)

Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Have Release Engineering added as a reviewer to any packaging PR
+* @daos-stack/build-and-release-watchers
```
Comment (liw) [inline .github/CODEOWNERS:2]: Any idea how to fix this "unknown owner" error?

Comment (brianjmurrell) [reply .github/CODEOWNERS:2]: Fixed.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
You need to update the `raft.spec` and `debian/changelog` so that when this lands it does not overwrite the existing packages in the repo.

**Comment by liw** [comment]
> You need to update the `raft.spec` and `debian/changelog` so that when this lands it does not overwrite the existing packages in the repo.

I've bumped the version.

**Review by kccain — APPROVED** [review]

File: raft.spec [hunk 2]
```diff
@@ -61,6 +61,11 @@ cp -a include/* %{buildroot}/%{_includedir}
 
 
 %changelog
+* Wed Apr 06 2022 Li Wei <wei.g.li@intel.com> -0.9.1-1
+- Fix membership changes
+- Fix node ID initialization
+- Update packaging
```
[the remark this answers is not in the document]
Comment (liw) [reply raft.spec:67]: OK.

**Pull request merged**